diagnostic.md

```diff
@@ -5,28 +5,37 @@ Record your responses inside the fenced code blocks below each question.
 1.  Give an example of a visual hierarchy that could be modeled with components.
 
     ```md
-    <!-- your response here -->
+    Visual Hierarchy:
+    ![visual hierarchy](http://emberigniter.com/images/distant-components/diagram.png)
+
+    Based on the url, the routing layer resolve a specific model to render the appropriate component.
+    The deeper the component is nested, the more levels of parents it needs to go up to send action to the parents of parents, etc and until it finally reach the route.
+
     ```
 
 1.  What is the command to generate a new component called '`my-map`'?
 
     ```sh
-    # your response here
+    ember g component my-map
     ```
 
 1.  What files are edited to produce a component, and what are their
     responsibilities?
 
     ```md
-    <!-- your response here -->
+    Once the component is generated, you can find the corresponding named folder inside components folder with a component.js file and a template.hbs file. The template file renders the content and the component.js defines the component behavior and can customize its html element.
```

Comment:
Yes.  The `template.hbs` is also responsible for binding data, and binding and dispatching actions.